domains/certificates/Query.API/API.AppTests/TransferTests.cs

```diff
@@ -0,0 +1,133 @@
+using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Threading.Tasks;
+using API.AppTests.Extensions;
+using API.AppTests.Helpers;
+using API.AppTests.Infrastructure;
+using API.AppTests.Mocks;
+using API.IntegrationTest.Infrastructure;
+using API.Query.API.ApiModels.Responses;
+using FluentAssertions;
+using MeasurementEvents;
+using Xunit;
+
+namespace API.AppTests;
+
+public sealed class TransferTests :
+    IClassFixture<QueryApiWebApplicationFactory>,
+    IClassFixture<MartenDbContainer>,
+    IClassFixture<RabbitMqContainer>,
+    IDisposable
+{
+    private readonly QueryApiWebApplicationFactory factory;
+    private readonly DataSyncWireMock dataSyncWireMock;
+
+    public TransferTests(
+        QueryApiWebApplicationFactory factory,
+        MartenDbContainer martenDbContainer,
+        RabbitMqContainer rabbitMqContainer)
+    {
+        dataSyncWireMock = new DataSyncWireMock(port: 9004);
```

Comment:
The instances that are injected here are marked with IClassFixture on the TransferTest-class. This means that it is the same instance that is injected in TransferTest. If we added DataSyncWireMock the same way, it means that it is the same instance and since each Fact/test method will setup what the mocked endpoint ("/meteringPoints" on DataSyncAPI) returns, we will end up with multiple registrations of the same mocked endpoint but expect different responses from the endpoint.
I do not know exactly how WireMock.NET handles this, but the safest approach is to create a new instance and only have one registration of the mocked endpoint.

This is also one of the things that is annoying with writing our integration tests at the moment. If we go with seeding data, then we do not need DataSyncWireMock in this test, thus fixing this particular issue.